Rubberduck.Parsing/Symbols/DeclarationFinder.cs

```diff
@@ -378,28 +378,90 @@ public IEnumerable<Declaration> DeclarationsWithType(DeclarationType type)
 
         public IEnumerable<Declaration> FindHandlersForWithEventsField(Declaration field)
         {
-            List<Declaration> result;
-            return _handlersByWithEventsField.Value.TryGetValue(field, out result) 
+            return _handlersByWithEventsField.Value.TryGetValue(field, out var result) 
                 ? result 
                 : Enumerable.Empty<Declaration>();
         }
 
+        /// <summary>
+        /// Finds all members of a class that are implementing the interface defined by the passed context.
+        /// </summary>
+        /// <param name="context">The ImplementsStmtContext to find member for, e.g. 'Implements IFoo`</param>
+        /// <returns>Members of the containing class that implement the interface.</returns>
         public IEnumerable<Declaration> FindInterfaceMembersForImplementsContext(VBAParser.ImplementsStmtContext context)
         {
-            List<Declaration> result;
-            return _membersByImplementsContext.Value.TryGetValue(context, out result)
+            return _membersByImplementsContext.Value.TryGetValue(context, out var result)
                 ? result
                 : Enumerable.Empty<Declaration>();
         }
 
+        /// <summary>
+        /// Finds the interface declaration for a QualifiedSelection contained by a statement similar to "Implements IClass1"
+        /// </summary>
+        /// <param name="selection">The QualifiedSelection to search.</param>
+        /// <returns>The selected interface if found, null if not found.</returns>
+        public ClassModuleDeclaration FindInterface(QualifiedSelection selection)
+        {
+            return FindAllUserInterfaces().FirstOrDefault(declaration => declaration.References
+                .Where(refrnce => refrnce.Context.GetAncestor<VBAParser.ImplementsStmtContext>() != null)
```

Comment:
disemvoweled lambda parameter

Reply:
Also AFAICT the `Where` and the `Any` can be merged into a single `Any`

Reply:
Fixed.  My indentation sucked there too.